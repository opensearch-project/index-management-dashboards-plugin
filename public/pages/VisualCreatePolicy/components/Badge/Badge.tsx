/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import { EuiText, EuiNotificationBadge, EuiTextColor } from "@elastic/eui";

interface BadgeProps {
  text: string;
  number: number;
  color?: "subdued" | "accent" | undefined;
}

const Badge = ({ text, number, color = "subdued" }: BadgeProps) => (
  <EuiText size="xs" textAlign="center">
    <EuiTextColor color={color}>
      <p>
        {text}{" "}
        <EuiNotificationBadge size="s" color={color}>
          {number}
        </EuiNotificationBadge>
      </p>
    </EuiTextColor>
  </EuiText>
);

export default Badge;
