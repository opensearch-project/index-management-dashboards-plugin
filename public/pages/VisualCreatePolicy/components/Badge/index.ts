/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import Badge from "./Badge";

export default Badge;
