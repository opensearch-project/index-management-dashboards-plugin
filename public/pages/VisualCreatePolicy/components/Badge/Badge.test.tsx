/*
 * SPDX-License-Identifier: Apache-2.0
 *
 * The OpenSearch Contributors require contributions made to
 * this file be licensed under the Apache-2.0 license or a
 * compatible open source license.
 *
 * Modifications Copyright OpenSearch Contributors. See
 * GitHub history for details.
 */

import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { render } from "@testing-library/react";
import Badge from "./Badge";

describe("<Badge /> spec", () => {
  it("renders the component", () => {
    const { container } = render(<Badge text="Some text" number={2} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
