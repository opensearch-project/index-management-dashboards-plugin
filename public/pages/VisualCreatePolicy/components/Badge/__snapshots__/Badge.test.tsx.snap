// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Badge /> spec renders the component 1`] = `
<div
  class="euiText euiText--extraSmall"
>
  <div
    class="euiTextAlign euiTextAlign--center"
  >
    <span
      class="euiTextColor euiTextColor--subdued"
    >
      <p>
        Some text
         
        <span
          class="euiNotificationBadge euiNotificationBadge--subdued"
        >
          2
        </span>
      </p>
    </span>
  </div>
</div>
`;
